agent/system_metrics.go

```diff
@@ -0,0 +1,275 @@
+package agent
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/evergreen-ci/evergreen/model/task"
+	"github.com/evergreen-ci/evergreen/rest/client"
+	"github.com/evergreen-ci/timber"
+	metrics "github.com/evergreen-ci/timber/system_metrics"
+	"github.com/mongodb/grip"
+	"github.com/mongodb/grip/recovery"
+	"github.com/pkg/errors"
+	"google.golang.org/grpc"
+)
+
+// DataFormat describes how the time series data is stored.
+type DataFormat int32
+
+// Valid DataFormat values.
+const (
+	DataFormatText DataFormat = 0
+	DataFormatFTDC DataFormat = 1
+	DataFormatBSON DataFormat = 2
+	DataFormatJSON DataFormat = 3
+	DataFormatCSV  DataFormat = 4
+)
+
+func (f DataFormat) validate() error {
+	switch f {
+	case DataFormatText, DataFormatFTDC, DataFormatBSON, DataFormatJSON, DataFormatCSV:
+		return nil
+	default:
+		return errors.New("invalid schema type specified")
+	}
+}
+
+// MetricCollector is an interface representing an object that can collect
+// a single system metric over a series of time steps.
+type MetricCollector interface {
+	// Name returns a string indicating the type of metric collected, such as "uptime".
+	Name() string
+
+	// Format returns the format of the collected data.
+	Format() metrics.DataFormat
+
+	// Collect returns the value of the collected metric when the function is called.
+	Collect() ([]byte, error)
+}
+
+// SystemMetricsCollector handles collecting an arbitrary set of system
+// metrics at a fixed interval and streaming them to cedar.
+type SystemMetricsCollector struct {
+	mu              sync.Mutex
+	streamWg        sync.WaitGroup
+	closeWg         sync.WaitGroup
```

Comment:
Can we get rid of the Wg suffix? You typically don't need to include the type in the field name.

If not, can it be WG instead of Wg?